.github/workflows/codeql-analysis.yml

```
@@ -23,23 +22,38 @@ jobs:
   CodeQL-Build:
     name: CodeQL
     runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+      pull-requests: write
+    
+    env:
+      GOPRIVATE: "github.com/onflow, github.com/axiomzen"
+    
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v3
+      uses: actions/checkout@v5
 
     - name: Set up Go
-      uses: actions/setup-go@v3
+      uses: actions/setup-go@v6
       with:
-        go-version: ${{ env.GO_VERSION }}
+        go-version-file: go.mod
         cache: true
 
+    - name: Tidy Go and mod vendor
+      run: go mod tidy && go mod vendor
+
     - name: Initialize CodeQL
-      uses: github/codeql-action/init@v2
+      uses: github/codeql-action/init@v4
       with:
         languages: go
+        queries: security-extended
 
     - name: Build
-      run: make build
+      run: CGO_ENABLED=0 go build -mod=vendor -tags=no_cgo ./...
```

This should keep using the Makefile
```suggestion
      run: make build
```